src/main/java/nl/lexemmens/podman/CopyMojo.java

```diff
@@ -0,0 +1,196 @@
+
+    private void copyImage(ServiceHub hub, String sourceImage, String targetImage) throws MojoExecutionException {
+        getLog().info(String.format("Copying image %s to %s...", sourceImage, targetImage));
+        hub.getSkopeoExecutorService().copy(sourceImage, targetImage);
+    }
+
+    private void performCopyUsingCatalogFile(ServiceHub hub) throws MojoExecutionException {
+        getLog().info("Using container-catalog.txt to perform Skopeo copy.");
+
+        // Use a customized repository session, setup to force a few behaviors we like.
+        DefaultRepositorySystemSession tempSession = new DefaultRepositorySystemSession(repositorySystemSession);
+        tempSession.setUpdatePolicy(RepositoryPolicy.UPDATE_POLICY_ALWAYS);
+
+        File tempRepo = null;
+        if (skopeo.getCopy().getDisableLocal()) {
+            getLog().info("Disabling local repository @ " + tempSession.getLocalRepository().getBasedir());
+            try {
+                tempRepo = Files.createTempDirectory("podman-maven-plugin-repo").toFile();
+
+                getLog().info("Using temporary local repository @ " + tempRepo.getAbsolutePath());
+                tempSession.setLocalRepositoryManager(localRepositoryManagerFactory.newInstance(tempSession, new LocalRepository(tempRepo)));
+            } catch (IOException | NoLocalRepositoryManagerException e) {
+                getLog().warn("Failed to disable local repository path.", e);
+            }
+        }
+        tempSession.setReadOnly();
+
+        List<RemoteRepository> remoteRepositories = getRemoteRepositories();
+        List<String> cataloguedImages = getCatalog(tempSession, remoteRepositories);
+        Map<String, String> transformedImages = performTransformation(cataloguedImages);
+
+        for (Map.Entry<String, String> imageEntry : transformedImages.entrySet()) {
+            copyImage(hub, imageEntry.getKey(), imageEntry.getValue());
+        }
+
+        if (skopeo.getCopy().getDisableLocal()) {
+            if (tempRepo != null) {
+                try {
+                    FileUtils.deleteDirectory(tempRepo);
+                } catch (IOException e) {
+                    getLog().warn("Failed to cleanup temporary repository directory: " + tempRepo);
+                }
+            }
+        }
+    }
+
+    private List<RemoteRepository> getRemoteRepositories() throws MojoExecutionException {
+        List<ArtifactRepository> remoteArtifactRepositories;
+        String sourceCatalogRepository = skopeo.getCopy().getSourceCatalogRepository();
+
+        if (sourceCatalogRepository == null) {
+            getLog().info("Using all remote repositories to find container catalog.");
+            remoteArtifactRepositories = project.getRemoteArtifactRepositories();
+        } else {
+            Optional<ArtifactRepository> repository = project.getRemoteArtifactRepositories()
+                    .stream()
+                    .filter(r -> r.getId().equals(sourceCatalogRepository))
```

Comment:
We could do with changing `r` to `repository`

Reply:
Will change it to `repo`, since `repository` conflicts with the var defined above.